src/App.tsx

```diff
@@ -1,22 +1,43 @@
-import { Route, Routes } from "react-router-dom";
-import { Main } from "./pages/Main";
+import { Route, Routes, useNavigate } from "react-router-dom";
+import { match } from "ts-pattern";
+import { useDebouncedCallback } from "use-debounce";
+import { useDeskproAppEvents, useDeskproAppClient } from "@deskpro/app-sdk";
 import { Redirect } from "./components/Redirect/Redirect";
 import { CreateNote } from "./pages/Create/Note";
 import { FindOrCreate } from "./pages/FindOrCreate/FindOrCreate";
 import { ViewTask } from "./pages/View/Task";
 import { EditTask } from "./pages/Edit/Task";
-import { VerifySettingsPage } from "./pages/VerifySettingsPage";
+import { HomePage, LoadingAppPage, VerifySettingsPage } from "@/pages";
+import { isNavigatePayload } from "@/utils";
+import type { EventPayload } from "@/types";
 
 const App = () => {
+  const navigate = useNavigate();
+  const { client } = useDeskproAppClient();
+
+  const debounceElementEvent = useDebouncedCallback((_, __, payload: EventPayload) => {
+    return match(payload.type)
+      .with("changePage", () => isNavigatePayload(payload) && navigate(payload.path))
+      .run();
+  }, 100);
+
+  useDeskproAppEvents({
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
We shouldn't need to ignore types if they're correct. What's the exception here?

Reply:
fixed